pkg/terra/config.go

```diff
@@ -0,0 +1,143 @@
+package terra
+
+import (
+	"sync"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/smartcontractkit/chainlink-terra/pkg/terra/db"
+)
+
+var DefaultConfigSet = ConfigSet{
+	// ~8s per block, so ~80s until we give up on the tx getting confirmed
+	// Anecdotally it appears anything more than 4 blocks would be an extremely long wait.
+	BlocksUntilTxTimeout:  10,
+	ConfirmMaxPolls:       100,
+	ConfirmPollPeriod:     time.Second,
+	FallbackGasPriceULuna: sdk.MustNewDecFromStr("0.01"),
+	GasLimitMultiplier:    1.5,
+	// The max gas limit per block is 1_000_000_000
+	// https://github.com/terra-money/core/blob/d6037b9a12c8bf6b09fe861c8ad93456aac5eebb/app/legacy/migrate.go#L69.
+	// The max msg size is 10KB https://github.com/terra-money/core/blob/d6037b9a12c8bf6b09fe861c8ad93456aac5eebb/x/wasm/types/params.go#L15.
+	// Our msgs are only OCR reports for now, which will not exceed that size.
+	// There appears to be no gas limit per tx, only per block, so theoretically
+	// we could include 1000 msgs which use up to 1M gas.
+	// To be conservative and since the number of messages we'd
+	// have in a batch on average roughly correponds to the number of terra ocr jobs we're running (do not expect more than 100),
+	// we can set a max msgs per batch of 100.
+	MaxMsgsPerBatch: 100,
+}
+
+type Config interface {
+	BlocksUntilTxTimeout() int64
+	ConfirmMaxPolls() int64
```

Comment:
actually I think we can simplify here since max polls can be derived from blocks until tx timeout and poll period. We can just specify those two then max polls = timeout blocks * time per block (upper bound) / poll period